### ISSUE TITLE
training and validation loss nan

### ISSUE THREAD

**Issue by Barfknecht** [nicolov/segmentation_keras#26]
First of all I just want to thank you for the great work.
I am having an issue during training, my loss and val_loss is nan, however I am still getting values for accuracy and val_acc. I am training on the PASCAL_VOC 2012 dataset with the segmentation class pngs. 
![rsz_screenshot_from_2017-08-09_17-24-13](https://user-images.githubusercontent.com/5606029/29129744-1734355a-7d28-11e7-87ba-356f9557043c.png)

keras 1.2.1 & 2.0.6
tensorflow-gpu 1.2.1
python 3.6.1

**Comment by nicolov** [comment]
Have you managed to fix this?

**Comment by Barfknecht** [comment]
Yes, the dataset was corrupt somehow. Just re-downloaded and it is fine.

**Comment by dimaxano** [comment]
@Barfknecht Could you, please, tell what exactly was wrong with your dataset, because I have this problem on my own dataset. But it doesn't look corrupted

**Comment by Barfknecht** [comment]
@dimaxano  Can you give an example of an image? 
The issue with mine was that the pixel values of the masks were not valid, so instead of them indicating the class, it had corrupted it and changed it to random ASCII characters.

**Comment by dimaxano** [comment]
![not_det_r_86c32981-6230-4f84-b4db-7429a1b37050_5216350389](https://user-images.githubusercontent.com/14369906/31709406-9715e68e-b3fa-11e7-9e64-fe71948a2ed1.jpg)
![not_det_r_86c32981-6230-4f84-b4db-7429a1b37050_5216350389](https://user-images.githubusercontent.com/14369906/31709407-97360f90-b3fa-11e7-8ccd-798806cc3a76.png)
Here is one of my image

**Comment by Barfknecht** [comment]
@dimaxano Well I can see your problem. So the issue is that the image mask has not been correctly formatted. So if you look in the benchmark_RELEASE/dataset/pngs folder you will see how the masks have been formatted. For example, 2008_000026.png is a mask of a person. That is class 15 of this altered pascal_voc dataset. When you open the png as a matrix, every pixel where the person is located has a pixel value of 15. So the input image is grayscale where the pixel values represent the class associated with it. So I have added one more class to this set, namely tyres. I have then made this my 21st class and when I created the mask, every pixel which is a tyre I set that pixel value to 21.

I hope this helps

**Comment by dimaxano** [comment]
@Barfknecht Thanks for advice! Now it works

**Comment by Barfknecht** [comment]
@dimaxano No problem, can you give me some advice on how you trained on your own dataset?

**Comment by dimaxano** [comment]
@Barfknecht For now, I have some troubles with predicting (https://github.com/nicolov/segmentation_keras/issues/33), but training  was quite easy (except this problem with labels ). It is more important to prepare data in a right way
